strawberryfields/gbs/g_sample.py

```diff
@@ -132,18 +131,7 @@ def to_subgraphs(graph: nx.Graph, samples: list) -> list:
     graph_nodes = list(graph.nodes)
     node_number = len(graph_nodes)
 
-    def to_subgraph(s: list) -> list:
-        """Convert a single sample to a subgraph.
-
-        Args:
-           s (list): a binary sample of ``len(nodes)``
-
-        Returns:
-            list: a subgraph specified by its nodes
-        """
-        return list(np.nonzero(s)[0])
-
-    subgraph_samples = [to_subgraph(s) for s in samples]
+    subgraph_samples = [sample.modes_from_counts(s) for s in samples]
```

Comment:
`to_subgraph` was an internal function of `to_subgraphs` which could take in one sample and return a subgraph specified by its nodes. E.g.: [1, 0, 1, 0, 1, 1] -> [0, 2, 4, 5]

In parallel we needed something in the `similarity` module that could take in one sample and return a list of modes where photons were counted. E.g., [1, 0, 2, 0, 3, 1] -> [0, 2, 2, 4, 4, 4, 5]

Since the functionality in the `similarity` module also encompassed `to_subgraph`, we decided to make this a single function sitting in the `sample` module (`sample.modes_from_counts`). This means that `to_subgraph` could be removed and replaced with the (tested and documented) `modes_from_counts`.